tests/always/test_project_structure.py

```diff
@@ -135,33 +133,123 @@ def filepath_to_module(filepath: str):
+
+    # Those operators are deprecated, so we do not need examples for them
+    DEPRECATED_OPERATORS: Set = set()
+
+    # Those operators should not have examples as they are never used standalone (they are abstract)
+    BASE_OPERATORS: Set = set()
+
+    # Please add the examples to those operators at the earliest convenience :)
+    MISSING_EXAMPLES_FOR_OPERATORS: Set = set()
+
+    def example_paths(self):
+        """Override this method if your example dags are located elsewhere"""
+        # old_design:
+        yield from glob.glob(
+            f"{ROOT_FOLDER}/airflow/providers/{self.PROVIDER}/**/example_dags/example_*.py", recursive=True
+        )
+        # new_design:
+        yield from glob.glob(
+            f"{ROOT_FOLDER}/tests/system/providers/{self.PROVIDER}/**/example_*.py", recursive=True
+        )
+
+    def test_missing_example_for_operator(self):
+        """
+        Assert that all operators defined under operators, sensors and transfers directories
+        are used in any of the example dags
+        """
+        all_operators = self.list_of_operators()
+        assert 0 != len(all_operators), "Failed to retrieve operators, override operator_paths if needed"
+        all_operators = set(all_operators.keys())
+        for example in self.example_paths():
+            all_operators -= get_imports_from_file(example)
+
+        covered_but_omitted = self.MISSING_EXAMPLES_FOR_OPERATORS - all_operators
+        all_operators -= self.MISSING_EXAMPLES_FOR_OPERATORS
+        all_operators -= self.DEPRECATED_OPERATORS
+        all_operators -= self.BASE_OPERATORS
+        assert set() == all_operators, (
+            "Not all operators are covered with example dags. "
+            "Update self.MISSING_EXAMPLES_FOR_OPERATORS if you want to skip this error"
+        )
+        assert set() == covered_but_omitted, "Operator listed in missing examples but is used in example dag"
+
+
+class AssetsCoverageTest(ProjectStructureTest):
+    """Checks that every operator have operator_extra_links attribute"""
+
+    # These operators should not have assets
+    ASSETS_NOT_REQUIRED: Set = set()
+
+    def test_missing_assets_for_operator(self):
+        all_operators = self.list_of_operators()
+        assets, no_assets = set(), set()
+        for name, operator in all_operators.items():
+            for attr in operator.body:
+                if (
+                    isinstance(attr, ast.Assign)
+                    and attr.targets
+                    and getattr(attr.targets[0], "id", "") == "operator_extra_links"
+                ):
+                    assets.add(name)
+                    break
+            else:
+                no_assets.add(name)
+
+        asset_should_be_missing = self.ASSETS_NOT_REQUIRED - no_assets
+        no_assets -= self.ASSETS_NOT_REQUIRED
+        # TODO: (bhirsz): uncomment when we reach full coverage
+        # assert set() == no_assets, "Operator is missing assets"
```

Comment:
Why not add a field akin to `MISSING_EXAMPLES_FOR_OPERATORS` so that this can be uncommitted while the work is being done to patch up the operators that are failing?

In fact, it looks like you're actually setting `MISSING_ASSETS_FOR_OPERATORS` below in the `TestGoogleProviderProjectStructure` class, but not using it here.

Reply:
That's actually my mistake - MISSING_ASSETS_FOR_OPERATORS should be named ASSETS_NOT_REQUIRED in Google test class. I can additionaly create MISSING_ASSETS_FOR_OPERATORS and do it the same way it's done for example coverage - I was hesistant since we have hundreds of those in Google :D Though I can update the check there and then override test method in Google with expected fail decorator and call it through the super statement.

Reply:
I've added ``MISSING_ASSETS_FOR_OPERATORS`` so it's possible to define list of missing assets but it's also possible to temporarily expect test to fail:

```
    @pytest.mark.xfail(reason="We did not reach full coverage yet")
    def test_missing_assets_for_operator(self):
        super().test_missing_assets_for_operator()
```
(example from the Google provider). When we will decrease number of missing assets I will override ``MISSING_ASSETS_FOR_OPERATORS`` with missing assets and expect test to pass.